backend/src/routes/__test__/goal.route.test.ts

```diff
@@ -0,0 +1,421 @@
+            .expect(httpStatus.BAD_REQUEST);
+
+        await supertest(app).get(`/api/goal/${goal._id}`)
+            .set('Authorization', token)
+            .expect(httpStatus.NO_CONTENT);
+
+        const { body: storedUser } = await supertest(app).get('/api/users')
+            .set('Authorization', token)
+            .expect(httpStatus.OK);
+        expect(storedUser).not.toContain(goal._id);
+
+    })
+})
+
+describe('GET /goal/:id', () => {
+    it ('Successfully retrieves an goal with a given json', async () => {
+        const newGoalId = await createUserGoal(token);
+
+        // retrieve the specified goal
+        const goal = await supertest(app)
+            .get(`/api/goal/${newGoalId}`)
+            .set('Accept', 'application/json')
+            .set('Authorization', token)
+            .expect(httpStatus.OK)
+
+        expect(goal._body.date_start === goal1Data.date_start)
+        expect(goal._body.date_end === goal1Data.date_end)
+        expect(goal._body.duration).toBe(goal1Data.duration)
+        expect(goal._body.recurring).toBe(goal1Data.recurring)
+
+    });
+
+    it('Fails when called with invalid goal object ID', async () => {
+        await createUserGoal(token)
+        // update an gpa; of id that does not exist
+        await supertest(app)
+            .get(`/api/goal/${123}`)
+            .set('Accept', 'application/json')
+            .set('Authorization', token)
+            .expect(httpStatus.NOT_FOUND)
+    });
+
+    it('Fails when goal object ID that does not exist', async () => {
+        // update an goal of id that does not exist
+        await supertest(app)
+            .get(`/api/goal/622b36166bb3a4e3a1ef62f1`)
+            .set('Accept', 'application/json')
+            .set('Authorization', token)
+            .expect(httpStatus.NOT_FOUND)
+    });
+    it('Fails when user does not exist for given token', async ()=>{
+        await supertest(app)
+            .get(`/api/goal/1234567890`)
+            .set('Accept', 'application/json')
+            .set('Authorization', token)
+            .expect(httpStatus.NOT_FOUND)
+    })
+    it('Fails when user is not authenticated', async ()=>{
+        await supertest(app)
+            .get(`/api/goal/1234567890`)
+            .set('Accept', 'application/json')
+            .expect(httpStatus.UNAUTHORIZED)
+    })
+});
+
+describe('PUT /goal/:id ', () => {
+    it('Successfully updates an goal with a given json', async () => {
+        const newGoalId = await createUserGoal(token);
+
+        // update a goal with 'goal3Data'
+        await supertest(app)
+            .put(`/api/goal/${newGoalId}`)
+            .set('Accept', 'application/json')
+            .set('Authorization', token)
+            .send(goal3Data)
+            .expect(httpStatus.NO_CONTENT) // since it's no content, need to get this goal to check updated fields
+
+        // retrieve the updated goal and compare
+        const updatedGoal = await supertest(app)
```

Comment:
I think you should retrieve it directly from the database, rather than calling other endpoints. (if get endpoint fails, this test will fail also)

Reply:
The reason for the GET was for security purposes when retrieving a goal from the database. The user needs authentication to retrieve their specific goal and not have full access to the database. This test mimics the authority required so it cannot retrieve the users' specific goal without a token.